EncounterScript_StreamlabsSystem.py

```diff
@@ -208,6 +211,105 @@ class Item:
     offence = 0
     defence = 0
 
+# ---------------------------------------
+# Functions used for the Gameplay Log File
+# ---------------------------------------
+
+def CreateGameplayLogFile():
+    create = open(LogFile, "w+")
+    create.close()
+    data2 = {}
+    unformattedDate = date.today()
+    data2['array'] = []
+    arr = {}
+    arr['date'] = unformattedDate.strftime("%d/%m/%Y")
+    arr['items'] = []
+    arr['encounters'] = []
+    arr['monsters'] = []
+    data2['array'].append(arr)
+
+    AddToFile(LogFile, data2)
+
+#------------------------------------------------------------------------------------------------------------------------
+
+# This function is used to log gameplay data. This is being used to debug and test the program. It will not be included
+#   for the entire duration of the project. It is in the release version, but just to continue the testing process.
+def AddLogEntry(section, data):
+    logFileData = ""
+    unformattedDate = date.today()
+    currentDate = unformattedDate.strftime("%d/%m/%Y")
+
+    with open(LogFile) as json_file:
+        logFileData = json.load(json_file)
+        DataAdded = False
+
+        for logEntry in logFileData['array']:
+            if logEntry['date'] == currentDate:
+                if section == "monsters":
+                    MonsterIsUnique = True
```

Comment:
Rather than having a bool for each section, create one bool called `EntryIsUnique`

Reply:
Removed individual booleans and added "EntryIsUnique"